Added guide for migrating VMs using CLI/kubectl

amar-chand opened the pull request:
## What this PR does / why we need it

The guide added in the PR can help people automate the migrations.

## Which issue(s) this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*

This PR fixes no issues.

## Special notes for your reviewer

No notes.

## Testing done

Checked the preivew of the document. 
 <div id='description'>
    <a href="https://bito.ai#summarystart"></a>
<h3>Summary by Bito</h3>
This pull request enhances GitHub workflows and documentation for improved release management, adding new triggers, input parameters, and an archiving mechanism to the release notes workflow. The deployment workflow now targets gh-pages branch, while documentation updates include mermaid integration and clearer VM migration guidance.
<!-- Disabling unit_tests and post_effort_to_review
<br>
<br>
<b>Unit tests added</b>: False
<br>
<br>
<b>Estimated effort to review (1-5, lower is better)</b>: 3
-->
</div>

amar-chand commented on docs/src/content/docs/guides/migrating_using_cli_and_kubectl.md:
```diff
@@ -0,0 +1,421 @@
+* [Gather Information](#gather-information)
+    * [Gather the credentials to VMware vCenter and OpenStack](#gather-the-credentials-to-vmware-vcenter-and-openstack)
+    * [Gather the details of the VM to be migrated](#gather-the-details-of-vms-to-be-migrated)
+    * [Gather the details of OpenStack networks and volumeTypes](gather-the-details-of-openstack-networks-and-volumetypes)
```
Updated the same as part of other commit

amar-chand commented on docs/src/content/docs/guides/migrating_using_cli_and_kubectl.md:
```diff
@@ -0,0 +1,421 @@
+  source:
+    vmwareRef: vcenter-a
+  destination:
+    openstackRef: openstack-a
```
Updated the same as part of other commit

amar-chand commented on docs/src/content/docs/guides/migrating_using_cli_and_kubectl.md:
```diff
@@ -0,0 +1,421 @@
+  storageMapping: storagemapping
+  osType: linuxGuest 
+  source:
+    vmwareRef: vcenter-a
```
Updated the same as part of other commit

amar-chand commented:
/windsurf-review

amar-chand commented on docs/src/content/docs/guides/migrating_using_cli_and_kubectl.md:
```diff
@@ -0,0 +1,421 @@
+    adminInitiatedCutOver: false
+    performHealthChecks: false
+    healthCheckPort: "443"
+  virtualmachines:
+    - - vm-1
```
The structure already added is correct.

amar-chand commented on docs/src/content/docs/guides/migrating_using_cli_and_kubectl.md:
```diff
@@ -0,0 +1,421 @@
+
+* [Gather Information](#gather-information)
+    * [Gather the credentials to VMware vCenter and OpenStack](#gather-the-credentials-to-vmware-vcenter-and-openstack)
+    * [Gather the details of the VM to be migrated](#gather-the-details-of-vms-to-be-migrated)
```
The anchor link has been updated.

amar-chand commented on docs/src/content/docs/guides/migrating_using_cli_and_kubectl.md:
```diff
@@ -0,0 +1,421 @@
+    * [Create the MigrationTemplate custom resource](#create-the-migrationtemplate-custom-resource)
+    * [Create the MigrationPlan custom resource](#create-the-migrationplan-custom-resource)
+* [Monitor the progress](#monitor-the-progress)
+    * [Checking Migration (custom resource)](#migration-custom-resource)
```
The anchor link is correct.

spai-p9 commented on docs/src/content/docs/guides/migrating_using_cli_and_kubectl.md:
```diff
@@ -0,0 +1,421 @@
+    adminInitiatedCutOver: false
+    performHealthChecks: false
+    healthCheckPort: "443"
+  virtualmachines:
```
```suggestion
  virtualMachines:
```
@amar-chand Here it should be camel casing.

amar-chand replied:
@spai-p9 Sure. Corrected it.

roopakparikh reviewed (changes requested):
Some nits, please make the changes before merging

OmkarDeshpande7 commented on docs/src/content/docs/guides/migrating_using_cli_and_kubectl.md:
```diff
@@ -0,0 +1,421 @@
+* VM's name using `spec.vms.name`
+* Get the datastores under `spec.vms.datastores`. This will be used for storage mapping.
+* Get the networks under `spec.vms.networks`. This will be used for network mapping.
+* (Optional) Get the Operating System type under `spec.vms.osType`. The reason for keeping it optional is that vJailbreak can auto-detect the OS most of the time, if the VM is turned on in VMware. For cold migration or migration of a VM that is in powered off state, this field will be required.
```
`spec.vms.osType` is now `spec.vms.osFamily`

OmkarDeshpande7 commented on docs/src/content/docs/guides/migrating_using_cli_and_kubectl.md:
```diff
@@ -0,0 +1,421 @@
+* Get the networks under `spec.vms.networks`. This will be used for network mapping.
+* (Optional) Get the Operating System type under `spec.vms.osType`. The reason for keeping it optional is that vJailbreak can auto-detect the OS most of the time, if the VM is turned on in VMware. For cold migration or migration of a VM that is in powered off state, this field will be required.
+
+**Note**: Make sure that the `spec.vms.vmState` is set to `running`, otherwise the migration will not take place.
```
Can we verify this?
From CLI, I think it will trigger migration even if `spec.vms.vmState` is not in `running`.
You just need to provide enough information.

amar-chand replied:
I verified this @OmkarDeshpande7.
If we use the CLI, the migration initiates even if the VM is in Powered OFF state.
Will remove this statement.

OmkarDeshpande7 commented on docs/src/content/docs/guides/migrating_using_cli_and_kubectl.md:
```diff
@@ -0,0 +1,421 @@
+    cpu: 1
+    memory: 4096
+    name: vm-1
+    osType: linuxGuest
```
```suggestion
    osFamily: linuxGuest
```

amar-chand replied:
Thanks @OmkarDeshpande7 . It has been updated.

OmkarDeshpande7 commented on docs/src/content/docs/guides/migrating_using_cli_and_kubectl.md:
```diff
@@ -0,0 +1,421 @@
+spec:
+  networkMapping: networkmapping 
+  storageMapping: storagemapping
+  osType: linuxGuest 
```
```suggestion
  osFamily: linuxGuest 
```

amar-chand commented on docs/src/content/docs/guides/migrating_using_cli_and_kubectl.md:
```diff
@@ -0,0 +1,421 @@
+
+#### Create the MigrationTemplate custom resource
+
+Now that we have `StorageMapping` and `NetworkMapping` resources in place, let’s create a `MigrationTemplate`, it will be referenced by `MigrationPlan` resources.
```
Sure @roopakparikh. Updated here:
https://github.com/amar-chand/vjailbreak/blob/d219d3af743a7e7ca531fb99a8100f1895791711/docs/src/content/docs/guides/migrating_using_cli_and_kubectl.md?plain=1#L293

amar-chand commented on docs/src/assets/information-flow.png:
@roopakparikh I've updated the image path. Here:
https://github.com/platform9/vjailbreak/pull/659/commits/4904d9ddbf6e6fad4d86fb26708f32f37fa472f7

amar-chand commented on docs/src/content/docs/guides/migrating_using_cli_and_kubectl.md:
```diff
@@ -0,0 +1,421 @@
+
+Before moving to the action, let’s understand the various resources that will be created to perform a migration, and how they are related to each other.
+![image](../../../assets/information-flow.png)
+
```
Thanks @roopakparikh. Updated. Here: 
https://github.com/platform9/vjailbreak/pull/659/commits/4904d9ddbf6e6fad4d86fb26708f32f37fa472f7

amar-chand commented on docs/src/content/docs/guides/migrating_using_cli_and_kubectl.md:
```diff
@@ -0,0 +1,421 @@
+title: "Migrate VMs using CLI and kubectl"
+description: "Learn how to automate the migration of Virtual Machines from VMware to PCD using vJailbreak"
+---
+vJailbreak comprises multiple Kubernetes Controllers which work on Custom Resources (CRs). When we perform the migrations through the UI, the UI itself takes care of creating these CRs for us, hence paving the way for migration to happen. In this tutorial, we will understand how we can migrate a VM using vJailbreak via CLI. These steps might be helpful if you are looking to automate the migration ...
```
Hi @roopakparikh 
Kindly clarify.

roopakparikh replied:
only "These steps might be helpful if you are looking to automate the migration."

OmkarDeshpande7 reviewed (approved):
lgtm

OmkarDeshpande7 commented:
@amar-chand this PR needs to have only one file change. Other file renames are unintentional. Please correct it

amar-chand commented:
Thank you for suggestions @OmkarDeshpande7 and @sharma-tapas
I've:
- reverted the commit which was causing the file location changes.
- moved the base branch to `gh-pages`.
